crates/torii/core/src/cache.rs

```diff
@@ -0,0 +1,56 @@
+use std::collections::HashMap;
+
+use dojo_types::schema::Ty;
+use sqlx::SqlitePool;
+use tokio::sync::RwLock;
+
+use crate::error::{Error, QueryError};
+use crate::model::{parse_sql_model_members, SqlModelMember};
+
+type ModelName = String;
+
+pub struct ModelCache {
+    pool: SqlitePool,
+    schemas: RwLock<HashMap<ModelName, Ty>>,
+}
+
+impl ModelCache {
```

Comment:
I think we will eventually need a way to invalidate a cache entry when we index an update to an existing model. Lets create an issue to track it